pkg/registry/apis/datasource/sub_resource.go

```diff
@@ -75,7 +75,7 @@ func (r *subResourceREST) Connect(ctx context.Context, name string, opts runtime
 			RawQuery: clonedReq.URL.RawQuery,
 		}
 
-		err = r.builder.client.CallResource(ctx, &backend.CallResourceRequest{
+		err = r.builder.handlers.CallResourceHandler.CallResource(ctx, &backend.CallResourceRequest{
```

Comment:
if we keep the current interface (`PluginClient`) but we remove the `backend.CallResourceHandler`, we could do here something like:

```go
		crHandler, ok := r.builder.client.(backend.CallResourceHandler)
		if !ok {
			responder.Error(fmt.Errorf("client does not implement backend.CallResourceHandler"))
			return
		}

		err = crHandler.CallResource(ctx, &backend.CallResourceRequest{
```

Reply:
it would... but this only gets created if the handler exists https://github.com/grafana/grafana/pull/84969/files#diff-4f8682b7549ec2d252d819509508d94153e4e75210d627b625ff95800af6f2dfR211